engine/execution/computation/computer/computer_test.go

```diff
@@ -947,3 +952,15 @@ func generateEvents(eventCount int, txIndex uint32) []flow.Event {
 	}
 	return events
 }
+
+func ensureEventsIndexSeq(t *testing.T, events []flow.Event, chainID flow.ChainID) {
+	expectedEventIndex := uint32(0)
+	for _, event := range events {
+		require.Equal(t, expectedEventIndex, event.EventIndex)
+		if testutil.IsServiceEvent(event, chainID) {
+			expectedEventIndex += 2
```

Comment:
In our code at https://github.com/onflow/flow-go/blob/master/fvm/environment/event_emitter.go#L154-L155, for service events we do double counting (original behavior), so the event after one service event will have index as `+2`. 
However if `AppendServiceEvent()` fails and `AppendEvent()` is skipped, then if `EmitEvent()` is called again, we can continue to have unique index values. If this use case can happen, we may want to keep the existing counting logic.

Reply:
This looks like a bug. Can you open an issue for it?

Reply:
filed #3393